Automattic/jetpack
# Introduce VideoPress content gating

Fixes https://github.com/Automattic/greenhouse/issues/1808

## Proposed changes:

* Alters the behavior of private videos when embedded within blocks/posts that restrict content to subscribers.
* ~Introduces a way for blocks to control the unauthorized message the video player will display.~
* The video iframe and it's files will not be available if a logged in user has no paid subscription, even if logged in.

### Other information:


## Jetpack product discussion

pxWta-1VK-p2

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:






* You need the Jetpack plugin to test this, as all Earn blocks are Jetpack-plugin-only.
* Do a full rebuild: `pnpm jetpack build plugins/jetpack --deps`
* Refer to pxWta-1VK-p2 on what to sandbox and what not.
* You will need to setup paid subscriptions and paid newsletter subscriptions.
* You will need 1 simple user, 1 newsletter sub user and 1 paid sub user.
* You will need to upgrade to a plan that includes more than one VideoPress videos. (because you are sandboxing store, you might need to do that again)
* On your test site, ensure you can access the paid content block.
* If you haven't, connect to stripe. If you sandboxed store, you can choose to use a "test" connection when on stripe's site.
* You are connected to test stripe now.

### Paid content block

* Go ahead and create a new post and add a "Paid Content" block.
* Assign a subscription plan.
* On the Subscriber view, create a VideoPress block and upload a video. Set privacy to private.
* Save.
* View as admin/editor. You can see the video.
* Try to view as a non-paid subscriber. You should not be able to view.
* View as a paid subscriber. You can view content and video.

### Newsletter

* Enable Newsletters (Jetpack -> Settings -> Newsletters). Create a newsletter subscription.
* Create a new post and set access to paid subscribers only.
* Add a videopress video, set to private. Save and publish.
* View as non-subscriber. You cant access anything.
* View as subscriber. You can see the video.

### Paywall

* Enable Newsletters (Jetpack -> Settings -> Newsletters). Create a newsletter subscription.
* Create post and upload a Video. Set it to public.
* Add a paywall block and upload another video. Set it to private.
* Publish post.
* View as guest and subscriber. You only see the video if a subscriber.
